prov/tcp/src/tcpx_cq.c

```diff
@@ -38,84 +38,11 @@
 #define TCPX_DEF_CQ_SIZE (1024)
 
 
-/*
- * Must hold lock protecting ep_list
- */
-void tcpx_progress(struct dlist_entry *ep_list, struct util_wait *wait)
+static void tcpx_cq_progress(struct util_cq *util_cq)
 {
-	struct ofi_epollfds_event events[MAX_POLL_EVENTS];
-	struct fid_list_entry *fid_entry;
-	struct util_wait_fd *wait_fd;
-	struct dlist_entry *item;
-	struct tcpx_ep *ep;
-	struct fid *fid;
-	uint32_t inevent, outevent, errevent;
-	int nfds, i;
-
-	wait_fd = container_of(wait, struct util_wait_fd, util_wait);
-
-	dlist_foreach(ep_list, item) {
```

Comment:
In the new code this is dlist_foreach_safe, since progress_rx can disable the ep if it's been disconnected, which removes the ep from progress lists.  Check existing code to see if this is correct or a similar fix is needed in the old code.  Look at adding asserts to list iterators to verify item is in the list.

Reply:
I checked this flow, and AFAICT there's not a problem using dlist_foreach() here.  We're holding the ep_list_lock and nothing in the call paths should remove the ep from the list.